File: Data Structures & Algorithms/subsets/submission-0.cpp
class Solution {
public:

    vector<int>Nums;
    
    void backtrack(int n, int idx, vector<int>& subset, vector<vector<int>>& res){
        if(idx>=n) {
            res.push_back(subset);
            return;
        }
        subset.push_back(Nums[idx]);
        cout<<"with "<<Nums[idx]<<endl;
        for(int i=0;i<subset.size();i++){
            cout<<subset[i]<<" ";
        } cout<<endl;
        backtrack(n, idx+1, subset, res);
        subset.pop_back();
        cout<<"NOT "<<Nums[idx]<<endl;
        for(int i=0;i<subset.size();i++){
            cout<<subset[i]<<" ";
        } cout<<endl;
        backtrack(n, idx+1, subset, res);

    }


    vector<vector<int>> subsets(vector<int>& nums) {
        Nums = nums;
        vector<vector<int>> res;
        vector<int> subset;
        backtrack(Nums.size(), 0, subset, res);
        return res;
    }
};

File: Data Structures & Algorithms/interleaving-string/submission-0.py
class Solution:
    def isInterleave(self, s1: str, s2: str, s3: str) -> bool:
        if len(s1)+len(s2) != len(s3):
            return False
        
        dp = {}

        def f(i,j,k):

            if k==len(s3):
                return (i==len(s1)) and (j==len(s2))
            
            if (i,j) in dp:
                return dp[(i,j)]
            
            res = False
            if i<len(s1) and s1[i]==s3[k]:
                res = f(i+1,j,k+1)
            if j<len(s2) and s2[j]==s3[k]:
                res = res or f(i,j+1,k+1)
            
            dp[(i,j)] = res
            return res


        

        return f(0,0,0)